service/measurement-server/README.md

```diff
@@ -0,0 +1,127 @@
+# Service: CCNP Measurement Server
+
+Measurements are key assets in the confidential computing world for user to verify and determine the trustworthiness of the environment and its underlying platform.
+Users can utilize different kinds of measurements to monitor or validate the operations performed within the platform.
+
+This service (which is actually a GRPC server) will provide support to fetch the measurements for confidential cloud native environments, including TDX RTMR measurements, TPM PCR measurements and TEE reports.
+
+
+
+## Introduction
+
+This service provides functionality to fetch measurements for confidential cloud native environments, both platform level(`PAAS` option) and container level (`SAAS` option). Using this service, user can fetch different
+categories of measurements including: TDX RTMR measurements (`TDX_RTMR` option), TPM PCR measurements (`TPM` option) and TEE reports from different vendors (`TEE_REPORT` option).
+Here shows the proto buf for the service:
+
+```
+enum TYPE {
+    PAAS = 0;
+    SAAS = 1;
+}
+
+enum CATEGORY {
+    TEE_REPORT = 0;
+    TPM = 1;
+    TDX_RTMR = 2;
+}
+
+message GetMeasurementRequest {
+    TYPE measurement_type = 1;
+    CATEGORY measurement_category = 2;
+    string report_data = 3;
+    int32 register_index = 4;
+
+}
+
+message GetMeasurementReply {
+    string measurement = 1;
+}
+
+service Measurement {
+    rpc GetMeasurement (GetMeasurementRequest) returns (GetMeasurementReply) {}
+}
+```
+
+To select different level and category of measurements, user shall send out request with different settings.
+User can find sample request in the Testing section.
+
+The collected measurements are returned as json string to the client.
+
+
+
+## Installation
+
+The Measurement service can be deployed as either DaemonSet or sidecar in different user scenarios.
+
+
+### Prerequisite
+
+To simplify the deployment process, we use Helm as the basic tool for deployment. Please install Helm following the [Helm official guide](https://helm.sh/docs/intro/install/).
+Also, the ccnp device plugin need to installed before the installation of measurement server. Please refer to its [deployment guide](../../device-plugin/ccnp-device-plugin/README.md) for installation.
+
+### Build docker image
+
+The dockerfile for the service can be found under `container/measurement-server` directory. Use the following command to build the image:
+
+```
+cd ../..
+docker build -t ccnp_measurement_server:0.1 -f container/measurement-server/Dockerfile .
+```
+> Note: if you are using containerd as the default runtime for kubernetes. Please remember to use the following command to import the image into containerd:
```

Comment:
if you are using containerd as the default runtime for kubernetes, don't forget to use the following commands to import the image into containerd first: